geoh5py/workspace/workspace.py

```diff
@@ -1214,24 +1214,21 @@ def register(self, entity: Entity | EntityType | PropertyGroup):
 
         :param entity: The entity to be registered.
         """
-        dict_entity: dict[uuid.UUID, ReferenceType]
         if isinstance(entity, EntityType):
-            dict_entity = self._types
+            weakref_utils.insert_once(self._types, entity.uid, entity)
         elif isinstance(entity, Group):
-            dict_entity = self._groups
+            weakref_utils.insert_once(self._groups, entity.uid, entity)
         elif isinstance(entity, Data):
-            dict_entity = self._data
+            weakref_utils.insert_once(self._data, entity.uid, entity)
         elif isinstance(entity, ObjectBase):
-            dict_entity = self._objects
+            weakref_utils.insert_once(self._objects, entity.uid, entity)
         elif isinstance(entity, PropertyGroup):
-            dict_entity = self._property_groups
+            weakref_utils.insert_once(self._property_groups, entity.uid, entity)
```

Comment:
Ok that' good. Unrelated but will fix the comment raised after GEOPY-1319

Reply:
yes, and not changing anything. Marking it as resolve